packages/core/contracts/cross-chain-oracle/chain-adapters/Optimism_ChildMessenger.sol

```diff
@@ -0,0 +1,92 @@
+ * @dev This contract's is ownable via the onlyCrossDomainAccount modifier, restricting ownership to the cross-domain
+ * parent messenger contract that lives on L1.
+ */
+contract Optimism_ChildMessenger is OVM_CrossDomainEnabled, ChildMessengerInterface {
+    // The only L2 contract that can send messages over the bridge via the messenger is the oracle spoke.
+    address public oracleSpoke;
+
+    // Messenger contract on the other side of the L1<->L2 bridge.
+    address public parentMessenger;
+
+    // Hard coded default gas limit for L1 transactions.
+    uint32 public defaultGasLimit = 5_000_000;
+
+    // TODO: import from optimism contracts when they release their latest version.
+    address internal constant L2_CROSS_DOMAIN_MESSENGER = 0x4200000000000000000000000000000000000007;
+
+    /**
+     * @notice Construct the Optimism_ChildMessenger contract.
+     * @param _parentMessenger The address of the L1 parent messenger. Acts as the "owner" of this contract.
+     */
+    constructor(address _parentMessenger) OVM_CrossDomainEnabled(L2_CROSS_DOMAIN_MESSENGER) {
+        parentMessenger = _parentMessenger;
+    }
+
+    /**
+     * @notice Changes the stored address of the Oracle spoke, deployed on L2.
+     * @dev The caller of this function must be the parent messenger, over the canonical bridge.
+     * @param newOracleSpoke address of the new oracle spoke, deployed on L2.
+     */
+    function setOracleSpoke(address newOracleSpoke) public onlyFromCrossDomainAccount(parentMessenger) {
+        oracleSpoke = newOracleSpoke;
+    }
+
+    /**
+     * @notice Changes the stored address of the parent messenger, deployed on L1.
+     * @dev The caller of this function must be the parent messenger, over the canonical bridge.
+     * @param newParentMessenger address of the new parent messenger, deployed on L1.
+     */
+    function setParentMessenger(address newParentMessenger) public onlyFromCrossDomainAccount(parentMessenger) {
+        parentMessenger = newParentMessenger;
+    }
+
+    /**
+     * @notice Changes the default gas limit that is sent along with transactions to Ethereum.
+     * @dev The caller of this function must be the parent messenger, over the canonical bridge.
+     * @param newDefaultGasLimit the new L1 gas limit to be set.
+     */
+    function setDefaultGasLimit(uint32 newDefaultGasLimit) public onlyFromCrossDomainAccount(parentMessenger) {
+        defaultGasLimit = newDefaultGasLimit;
+    }
+
+    /**
+     * @notice Sends a message to the parent messenger via the canonical message bridge.
+     * @dev The caller must be the OracleSpoke on L2. No other contract is permissioned to call this function.
+     * @dev The L1 target, the parent messenger, must implement processMessageFromChild to consume the message.
+     * @param data data message sent to the L1 messenger. Should be an encoded function call or packed data.
+     */
+    function sendMessageToParent(bytes memory data) public override {
```

Comment:
We should emit an event here since this is an important state changing function that interacts with an external contract